internal/backends/nodejs/nodejs.go

```diff
@@ -309,6 +309,67 @@ var NodejsYarnBackend = api.LanguageBackend{
 	Guess:        nodejsGuess,
 }
 
+var NodejsPNPMBackend = api.LanguageBackend{
+	Name:             "nodejs-pnpm",
+	Specfile:         "package.json",
+	Lockfile:         "pnpm-lock.yaml",
+	FilenamePatterns: nodejsPatterns,
+	Quirks: api.QuirksAddRemoveAlsoLocks |
+		api.QuirksAddRemoveAlsoInstalls |
+		api.QuirksLockAlsoInstalls,
+	GetPackageDir: func() string {
+		return "node_modules"
+	},
+	Search: nodejsSearch,
+	Info:   nodejsInfo,
+	Add: func(pkgs map[api.PkgName]api.PkgSpec, projectName string) {
+		if !util.Exists("package.json") {
+			util.RunCmd([]string{"npm", "init", "-y"})
+		}
+		cmd := []string{"pnpm", "add"}
+		for name, spec := range pkgs {
+			arg := string(name)
+			if spec != "" {
+				arg += "@" + string(spec)
+			}
+			cmd = append(cmd, arg)
+		}
+		util.RunCmd(cmd)
+	},
+	Remove: func(pkgs map[api.PkgName]bool) {
+		cmd := []string{"pnpm", "remove"}
+		for name := range pkgs {
+			cmd = append(cmd, string(name))
+		}
+		util.RunCmd(cmd)
+	},
+	Lock: func() {
+		util.RunCmd([]string{"pnpm", "install"})
+	},
```

Comment:
Maybe we can `pnpm install --lockfile-only` and remove the LockAlsoInstalls quirk?

Reply:
Actually this quirk is a good thing since it cuts extra calls to the package manager. Let's keep it as it is.

Reply:
agreed, `LockAlsoInstalls` is good quirk :)